### PR TITLE
Mrn release

### PR DESCRIPTION
**Pull request by robertbutacu**

### CODE REVIEW

File: app/uk/gov/hmrc/ctc/models/api/MrnReleaseRequest.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ctc.models.api
+
+import play.api.libs.json.{Format, Json}
+import uk.gov.hmrc.ctc.models.data.TransitRecord
+
+case class MrnReleaseRequest(providerId: String, transit: TransitRecord)
```
Comment (peppe-hmrc) [inline app/uk/gov/hmrc/ctc/models/api/MrnReleaseRequest.scala]: is this needed? as far I know it should not be needed anymore

Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/models/api/MrnReleaseRequest.scala]: this is not needed

File: app/uk/gov/hmrc/ctc/repositories/TransitMongoQueryHelper.scala [hunk 2]
```diff
@@ -110,6 +112,20 @@ trait TransitMongoQueryHelper {
...
+  def releaseMrn(): JsObject = {
+    Json.obj(fields =
+      "$set" -> Json.obj(fields =
+        setStatusReady()),
+      "$unset" -> Json.obj(fields =
+        removeProviderId(),
+        removeExpirationTime()))
+  }
+
+  def removeProviderId(): (String, JsValueWrapper) = "processingInfo.providerId" -> 1
```
Comment (peppe-hmrc) [inline app/uk/gov/hmrc/ctc/repositories/TransitMongoQueryHelper.scala]: Why `1`? this is an option should it be None? I was trying with JsValue(None)

Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/repositories/TransitMongoQueryHelper.scala]: oh I see, ok fine nevermind :)

File: app/uk/gov/hmrc/ctc/repositories/TransitRepository.scala [hunk 3]
```diff
@@ -75,13 +77,25 @@ class TransitRepository @Inject()(mongo: ReactiveMongoComponent)(implicit execut
...
 
+  //TODO modify this as below, so its more useful in testing and also open to extension
   def find(mrn: MovementReferenceNumber): Future[Either[CTCError, TransitId]] = {
     find(selectMRN(mrn)).map {
       case h :: t =>  Right(TransitId(h.id))
       case Nil => Left(MongoDBNoResults("No results found"))
     }
   }
 
+  def findAllRecord(mrn: MovementReferenceNumber): Future[Either[CTCError, Transit]] = {
```
Comment (peppe-hmrc) [inline app/uk/gov/hmrc/ctc/repositories/TransitRepository.scala]: are these used??

File: app/uk/gov/hmrc/ctc/models/api/MrnReleaseRequest.scala (hunk 1 above)
Comment (robertbutacu) [reply app/uk/gov/hmrc/ctc/models/api/MrnReleaseRequest.scala]: removed

File: app/uk/gov/hmrc/ctc/repositories/TransitRepository.scala (hunk 3 above)
Comment (robertbutacu) [reply app/uk/gov/hmrc/ctc/repositories/TransitRepository.scala]: removed

**Pull request closed without merging**